src/test/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketAuthorMessageTest.java

```diff
@@ -0,0 +1,211 @@
+package com.cloudbees.jenkins.plugins.bitbucket;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.*;
+import static org.mockito.Mockito.*;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+
+import org.hamcrest.CoreMatchers;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketApi;
+import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketBranch;
+import com.cloudbees.jenkins.plugins.bitbucket.client.BitbucketIntegrationClientFactory;
+import com.cloudbees.jenkins.plugins.bitbucket.endpoints.BitbucketCloudEndpoint;
+
+import hudson.Extension;
+import hudson.Util;
+import jenkins.plugins.git.AbstractGitSCMSource.SCMRevisionImpl;
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.SCMHeadObserver;
+import jenkins.scm.api.SCMRevision;
+import jenkins.scm.api.trait.SCMHeadFilter;
+import jenkins.scm.api.trait.SCMSourceContext;
+import jenkins.scm.api.trait.SCMSourceRequest;
+import jenkins.scm.api.trait.SCMSourceTrait;
+import jenkins.scm.api.trait.SCMSourceTraitDescriptor;
+
+public class BitbucketAuthorMessageTest {
```

Comment:
May need to rework the testing strategy so that the changes to this repo can be verified here. (I am only giving a fly-by review so I have not deep dived)

Reply:
I do not understand. The test strategy does not go online, it's all offline.
If you mean that should be possible add use case (more commits/branches/PR) to the sample repository to provide more payloads as resource than I can zip and commit the repository (it's very small) as resource. If I miss the point please explain.

Reply:
As I said, fly by review. Answer this question: does the test rely on the trait class being available? If yes, then you’ll need to rework the tests when the trait is moved out. If no, then all good

Reply:
This test case only cover if message and author are available to any context filter, it is not related to the trait